### PR TITLE
docs: add microsyntax details

### PR DESCRIPTION
**Pull request by kapunahelewong**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Adds detailed information about microsyntax.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Here are the tables added to Structural Directives.

image

image

image

### CODE REVIEW

**Review by gkalpak — COMMENTED** [review]
OOC, why use the `:`-prefix? It is is a little confusing given that `:` can be part of DSL.

File: aio/content/guide/structural-directives.md [hunk 1]
```diff
@@ -275,53 +275,179 @@ In this example, the `[ngClass]="odd"` stays on the `<div>`.
...
 
 * The `let` keyword declares a [_template input variable_](guide/structural-directives#template-input-variable)
 that you reference within the template. The input variables in this example are `hero`, `i`, and `odd`.
-The parser translates `let hero`, `let i`, and `let odd` into variables named,
+The parser translates `let hero`, `let i`, and `let odd` into variables named
 `let-hero`, `let-i`, and `let-odd`.
 
-* The microsyntax parser takes `of` and `trackBy`, title-cases them (`of` -> `Of`, `trackBy` -> `TrackBy`),
-and prefixes them with the directive's attribute name (`ngFor`), yielding the names `ngForOf` and `ngForTrackBy`.
+* The microsyntax parser title-cases `of` and `trackBy` (`of` -> `Of`, `trackBy` -> `TrackBy`)
```
Comment (mhevery) [inline aio/content/guide/structural-directives.md]: This makes it sounds that we only title-cale `of` and `trackBy` We do it for every directive. Could this be more generic?

File: aio/content/guide/structural-directives.md [hunk 2]
```diff
@@ -275,53 +275,179 @@ In this example, the `[ngClass]="odd"` stays on the `<div>`.
...
+and prefixes them with the directive's attribute name (`ngFor`), which
+yields the names `ngForOf` and `ngForTrackBy`.
 Those are the names of two `NgFor` _input properties_ .
 That's how the directive learns that the list is `heroes` and the track-by function is `trackById`.
 
 * As the `NgFor` directive loops through the list, it sets and resets properties of its own _context_ object.
-These properties include `index` and `odd` and a special property named `$implicit`.
+These properties include `index`, `odd`, and a special property named `$implicit`.
```
Comment (mhevery) [inline aio/content/guide/structural-directives.md]: Those are examples, not limiting set of what parser understands.

File: aio/content/guide/structural-directives.md (hunk 1 above)
Comment (kapunahelewong) [reply aio/content/guide/structural-directives.md]: Updated to:

>The microsyntax parser title-cases all directives and prefixes them with the directive's
attribute name, such as `ngFor`. For example, the `ngFor` input properties,
`of` and `trackBy` become `ngForOf` and `ngForTrackBy`, respectively.

Is this more accurate @mhevery?

File: aio/content/guide/structural-directives.md (hunk 2 above)
Comment (kapunahelewong) [reply aio/content/guide/structural-directives.md]: Updated to:

>These properties can include, but aren't limited to, `index`, `odd`, and a special property named `$implicit`.

File: aio/content/guide/structural-directives.md (hunk 1 above)
Comment (mhevery) [reply aio/content/guide/structural-directives.md]: 👍

File: aio/content/guide/structural-directives.md (hunk 2 above)
Comment (mhevery) [reply aio/content/guide/structural-directives.md]: 👍

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**